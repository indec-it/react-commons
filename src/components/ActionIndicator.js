import React from 'react';
import PropTypes from 'prop-types';

import LoadingIndicator from './LoadingIndicator';
import SaveIndicator from './SaveIndicator';

const ActionIndicator = ({
    saving,
    savingLabel,
    loading,
    loadingLabel
}) => (
    (saving || loading) && (
        <div>
            <div
                className="modal fade show"
                style={{
                    display: 'block',
                    alignItems: 'center',
                    justifyContent: 'center'
                }}
            >
                <div
                    className="align-middle text-center h-100"
                    style={{
                        alignItems: 'center',
                        justifyContent: 'center',
                        display: 'flex'
                    }}
                >
                    {loading && loadingLabel}
                    {saving && savingLabel}
                </div>
            </div>
            <div className="modal-backdrop fade show"/>
        </div>
    )
);

ActionIndicator.propTypes = {
    saving: PropTypes.bool,
    loading: PropTypes.bool,
    loadingLabel: PropTypes.oneOf([
        PropTypes.string,
        PropTypes.instanceOf(React)
    ]),
    savingLabel: PropTypes.oneOf([
        PropTypes.string,
        PropTypes.instanceOf(React)
    ])
};

ActionIndicator.defaultProps = {
    saving: false,
    loading: false,
    loadingLabel: LoadingIndicator,
    savingLabel: SaveIndicator
};

export default ActionIndicator;
