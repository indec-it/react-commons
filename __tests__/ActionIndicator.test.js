import React from 'react';
import {shallow} from 'enzyme';

import ActionIndicator from '../src/components/ActionIndicator';

const renderComponent = (props = {}) => {
    const wrapper = shallow(<ActionIndicator {...props}/>);
    wrapper.indicator = () => wrapper.find('Fragment div div').children();
    return wrapper;
};

describe('<ActionIndicator/>', () => {
    it('should render the component if has prop saving', () => {
        const wrapper = renderComponent({saving: true});
        expect(wrapper.isEmptyRender()).toBe(false);
    });
    it('should render the component if has prop loading', () => {
        const wrapper = renderComponent({loading: true});
        expect(wrapper.isEmptyRender()).toBe(false);
    });
    it('should be empty component if doesnt exist prop loading or saving', () => {
        const wrapper = renderComponent();
        expect(wrapper.isEmptyRender()).toBe(true);
    });
    it('should render LoadingIndicator component if loading flag exists', () => {
        const wrapper = renderComponent({loading: true});
        expect(wrapper.indicator()).toMatchSnapshot();
    });
    it('should render SaveIndicator component if saving flag exists', () => {
        const wrapper = renderComponent({saving: true});
        expect(wrapper.indicator()).toMatchSnapshot();
    });
});
