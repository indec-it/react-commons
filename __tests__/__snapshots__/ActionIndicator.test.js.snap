// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionIndicator/> should render LoadingIndicator component if loading flag exists 1`] = `
<LoadingIndicator
  iconSize="2x"
  label="Cargando..."
/>
`;

exports[`<ActionIndicator/> should render SaveIndicator component if saving flag exists 1`] = `<SaveIndicator />`;
